bitcoin/src/blockdata/witness.rs

```diff
@@ -147,19 +147,15 @@ crate::internal_macros::define_extension_trait! {
         ///
         /// See [`Script::is_p2tr`] to check whether this is actually a Taproot witness.
         fn tapscript(&self) -> Option<&Script> {
-            self.last().and_then(|last| {
-                // From BIP341:
-                // If there are at least two witness elements, and the first byte of
-                // the last element is 0x50, this last element is called annex a
-                // and is removed from the witness stack.
-                if self.len() >= 3 && last.first() == Some(&TAPROOT_ANNEX_PREFIX) {
-                    self.nth(self.len() - 3).map(Script::from_bytes)
-                } else if self.len() >= 2 {
-                    self.nth(self.len() - 2).map(Script::from_bytes)
-                } else {
-                    None
-                }
-            })
+            if self.is_empty() {
```

Comment:
I think this self.is_empty(), as well as the one in taproot_control_block can be removed, since nth_to_last will always return None if we're empty, so it doesn't change the behavior of the function.

Reply:
You are 100% correct, I think the explicitness is worth the extra lines of code. Its just my style, I like things to be brain dead easy to read. Without this, for an empty witness stack, to work out what happens, one has to follow right through and think that `second_to_last` in the else clause will return `None` - ugh!